druid/src/widget/container.rs

```diff
@@ -135,29 +135,35 @@ impl<T: Data> Widget<T> for Container<T> {
     }
 
     fn paint(&mut self, paint_ctx: &mut PaintCtx, data: &T, env: &Env) {
-        let panel = RoundedRect::from_origin_size(
-            Point::ORIGIN,
-            paint_ctx.size().to_vec2(),
-            self.corner_radius,
-        );
-
-        if let Err(e) = paint_ctx.save() {
-            log::error!("{}", e);
-            return;
-        }
-
-        paint_ctx.clip(panel);
-
         if let Some(background) = self.background.as_mut() {
+            let bg_rect = RoundedRect::from_origin_size(
+                Point::ORIGIN,
+                paint_ctx.size().to_vec2(),
+                self.corner_radius,
+            );
+
+            if let Err(e) = paint_ctx.save() {
```

Comment:
I really get annoyed by this api, will look into a helper method (this is unrelated to the current  PR)

Reply:
Yeah the `with_save` method will reduce human failure chance.